File: leetcode/js/apply-operations-to-an-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var applyOperations = function (nums) {
  const n = nums.length;

  for (let i = 0; i < n - 1; i++) {
    if (nums[i] == nums[i + 1]) {
      nums[i] *= 2;
      nums[i + 1] = 0;
    }
  }

  const filtered = nums.filter((num) => num !== 0);
  const diff = n - filtered.length;

  for (let i = 0; i < diff; i++) {
    filtered.push(0);
  }

  return filtered;
};

File: leetcode/rust/apply-operations-to-an-array.rs
impl Solution {
    pub fn apply_operations(mut nums: Vec<i32>) -> Vec<i32> {
        for i in 0..nums.len() - 1 {
            if nums[i] == nums[i + 1] {
                nums[i] *= 2;
                nums[i + 1] = 0;
            }
        }

        let mut filtered: Vec<i32> = nums.clone().into_iter().filter(|&n| n != 0).collect();
        let diff = nums.len() - filtered.len();

        for i in 0..diff {
            filtered.push(0);
        }

        filtered
    }
}
